akka-javasdk/src/main/java/akka/javasdk/agent/Message.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.agent;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "@type")
+@JsonSubTypes({
+    @JsonSubTypes.Type(value = UserMessage.class, name = "UM"),
+    @JsonSubTypes.Type(value = AiMessage.class, name = "AIM")})
+public interface Message {
```

Comment:
Message sounds a bit too open to me, even when it is in the `agent` namespace. Could it be more specific?

Also, sealed in Java can work with multiple classes from separate files, you just have to list the allowed subtypes here. 

```java
public sealed class Shape
    permits Circle, Square, Rectangle {
}
```

Reply:
ExchangeMessage? ConversationMessage? I'm not too confident on the current naming as well to be honest.. I will think a bit more.